core/templates/pages/community-dashboard-page/services/contribution-opportunities-backend-api.service.ts

```diff
@@ -86,7 +89,12 @@ export class ContributionOpportunitiesBackendApiService {
           opportunityType, data.opportunities[index]));
       }
       if (successCallback) {
-        successCallback(opportunities, data.next_cursor, data.more);
+        const result = {
+          opportunities: opportunities,
+          nextCursor: data.next_cursor,
+          more: data.more
+        };
+        successCallback(result);
```

Comment:
Have you tested the UI manually after this change?

Reply:
@DubeySandeep Not really, need to find out where it is.

Reply:
@DubeySandeep How can I get to the contribution opportunity in UI?

Reply:
Ah, I was expecting that you have already tested this, please make sure to manually test the UI (In general) before creating PR!

Follow these steps:
1. Create an account and add yourself as an admin through the ROLES tab of the admin page.
2. Generate new structure dummy data from the bottom of the ACTIVITY tab of admin page.
3. Go to the JOBS tab and run the "ExplorationOpportunitySummaryModelRegenerationJob" job. [This step is needed as we expect the dummy data to create opportunity but as of now it's not creating as such so we regenerate it.]


Other way you can do:
1. Create and publish exploration and copy the expID from the URL bar
2. Create a Topic
3. Add a story in the topic
4. Add a chapter in the story and past the expID for the chapter exploration
5. Save the changes
6. Go to /community_dashboard page and check the "Translate text" tab

Reply:
@DubeySandeep I took the steps and got successful requests for `fetchSkillOpportunities` and `fetchTranslationOpportunities`, however, I noticed that I can't upload a voiceover nor svg image for the story and chapter and I get these errors:
<img width="1146" alt="Screenshot 2020-05-07 at 17 30 57" src="https://user-images.githubusercontent.com/11369764/81313736-8a644300-9088-11ea-9008-7d3b1e984931.png">

Reply:
Can you please check the server log, I think you can find the failure stack trace there? Let me know if that's not helpful? [Also, it would be great if you can explain what you tried to debug the issue and your findings!]

Reply:
@DubeySandeep I tried again to upload an SVG which again didn't work. This is what's getting printed in the server log:
```
ERROR    2020-06-11 18:48:44,035 base.py:476] Exception raised: AccessTokenRefreshError(u'invalid_grant: Bad Request',)
INFO     2020-06-11 18:48:44,299 module.py:835] default: "POST /createhandler/imageupload/story/RHhM5bpZumPg HTTP/1.1" 500 93
INFO     2020-06-11 18:49:00,146 client.py:755] Refreshing access_token
INFO     2020-06-11 18:49:00,382 client.py:777] Failed to retrieve access token: {
  "error": "invalid_grant",
  "error_description": "Bad Request"
}
ERROR    2020-06-11 18:49:00,390 api_server.py:364] Exception while handling app_identity_service.GetAccessToken()
Traceback (most recent call last):
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia_tools/google_appengine_1.9.67/google_appengine/google/appengine/tools/devappserver2/api_server.py", line 332, in _handle_POST
    api_response = _execute_request(request).Encode()
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia_tools/google_appengine_1.9.67/google_appengine/google/appengine/tools/devappserver2/api_server.py", line 188, in _execute_request
    make_request()
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia_tools/google_appengine_1.9.67/google_appengine/google/appengine/tools/devappserver2/api_server.py", line 183, in make_request
    request_id)
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia_tools/google_appengine_1.9.67/google_appengine/google/appengine/api/apiproxy_stub.py", line 151, in MakeSyncCall
    method(request, response)
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia_tools/google_appengine_1.9.67/google_appengine/google/appengine/api/app_identity/app_identity_defaultcredentialsbased_stub.py", line 197, in _Dynamic_GetAccessToken
    token = credentials.get_access_token()
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia_tools/google_appengine_1.9.67/google_appengine/lib/oauth2client_devserver/oauth2client/client.py", line 642, in get_access_token
    self.refresh(http)
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia_tools/google_appengine_1.9.67/google_appengine/lib/oauth2client_devserver/oauth2client/client.py", line 559, in refresh
    self._refresh(http.request)
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia_tools/google_appengine_1.9.67/google_appengine/lib/oauth2client_devserver/oauth2client/client.py", line 728, in _refresh
    self._do_refresh_request(http_request)
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia_tools/google_appengine_1.9.67/google_appengine/lib/oauth2client_devserver/oauth2client/client.py", line 790, in _do_refresh_request
    raise AccessTokenRefreshError(error_msg)
AccessTokenRefreshError: invalid_grant: Bad Request

WARNING  2020-06-11 18:49:00,404 tasklets.py:468] suspended generator _make_token_async(rest_api.py:103) raised RuntimeError(AccessTokenRefreshError(u'invalid_grant: Bad Request',))
WARNING  2020-06-11 18:49:00,404 tasklets.py:468] suspended generator get_token_async(rest_api.py:214) raised RuntimeError(AccessTokenRefreshError(u'invalid_grant: Bad Request',))
WARNING  2020-06-11 18:49:00,405 tasklets.py:468] suspended generator urlfetch_async(rest_api.py:247) raised RuntimeError(AccessTokenRefreshError(u'invalid_grant: Bad Request',))
WARNING  2020-06-11 18:49:00,405 tasklets.py:468] suspended generator run(api_utils.py:164) raised RuntimeError(AccessTokenRefreshError(u'invalid_grant: Bad Request',))
WARNING  2020-06-11 18:49:00,405 tasklets.py:468] suspended generator do_request_async(rest_api.py:190) raised RuntimeError(AccessTokenRefreshError(u'invalid_grant: Bad Request',))
WARNING  2020-06-11 18:49:00,405 tasklets.py:468] suspended generator do_request_async(storage_api.py:128) raised RuntimeError(AccessTokenRefreshError(u'invalid_grant: Bad Request',))
INFO     2020-06-11 18:49:00,408 base.py:466] Traceback (most recent call last):
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia_tools/google_appengine_1.9.67/google_appengine/lib/webapp2-2.5.2/webapp2.py", line 570, in dispatch
    return method(*args, **kwargs)
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia/core/controllers/acl_decorators.py", line 2640, in test_can_edit_entity
    return can_edit_story(reduced_handler)(self, entity_id, **kwargs)
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia/core/controllers/acl_decorators.py", line 1849, in test_can_edit_story
    return handler(self, story_id, **kwargs)
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia/core/controllers/acl_decorators.py", line 2623, in <lambda>
    arg_swapped_handler = lambda x, y, z: handler(y, x, z)
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia/core/controllers/editor.py", line 689, in post
    if fs.isfile(filepath):
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia/core/domain/fs_domain.py", line 283, in isfile
    return self._impl.isfile(filepath)
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia/core/domain/fs_domain.py", line 137, in isfile
    return bool(cloudstorage.stat(gcs_file_url, retry_params=None))
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia/third_party/gae-cloud-storage-1.9.22.1/cloudstorage/cloudstorage_api.py", line 149, in stat
    api_utils._quote_filename(filename))
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia/third_party/gae-cloud-storage-1.9.22.1/cloudstorage/rest_api.py", line 56, in sync_wrapper
    return future.get_result()
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia_tools/google_appengine_1.9.67/google_appengine/google/appengine/ext/ndb/tasklets.py", line 383, in get_result
    self.check_success()
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia_tools/google_appengine_1.9.67/google_appengine/google/appengine/ext/ndb/tasklets.py", line 427, in _help_tasklet_along
    value = gen.throw(exc.__class__, exc, tb)
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia/third_party/gae-cloud-storage-1.9.22.1/cloudstorage/storage_api.py", line 128, in do_request_async
    deadline=deadline, callback=callback)
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia_tools/google_appengine_1.9.67/google_appengine/google/appengine/ext/ndb/tasklets.py", line 427, in _help_tasklet_along
    value = gen.throw(exc.__class__, exc, tb)
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia/third_party/gae-cloud-storage-1.9.22.1/cloudstorage/rest_api.py", line 190, in do_request_async
    follow_redirects=False)
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia_tools/google_appengine_1.9.67/google_appengine/google/appengine/ext/ndb/tasklets.py", line 427, in _help_tasklet_along
    value = gen.throw(exc.__class__, exc, tb)
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia/third_party/gae-cloud-storage-1.9.22.1/cloudstorage/api_utils.py", line 164, in run
    result = yield tasklet(**kwds)
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia_tools/google_appengine_1.9.67/google_appengine/google/appengine/ext/ndb/tasklets.py", line 427, in _help_tasklet_along
    value = gen.throw(exc.__class__, exc, tb)
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia/third_party/gae-cloud-storage-1.9.22.1/cloudstorage/rest_api.py", line 247, in urlfetch_async
    self.token = yield self.get_token_async()
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia_tools/google_appengine_1.9.67/google_appengine/google/appengine/ext/ndb/tasklets.py", line 427, in _help_tasklet_along
    value = gen.throw(exc.__class__, exc, tb)
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia/third_party/gae-cloud-storage-1.9.22.1/cloudstorage/rest_api.py", line 214, in get_token_async
    self.scopes, self.service_account_id)
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia_tools/google_appengine_1.9.67/google_appengine/google/appengine/ext/ndb/tasklets.py", line 427, in _help_tasklet_along
    value = gen.throw(exc.__class__, exc, tb)
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia/third_party/gae-cloud-storage-1.9.22.1/cloudstorage/rest_api.py", line 103, in _make_token_async
    token, expires_at = yield rpc
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia_tools/google_appengine_1.9.67/google_appengine/google/appengine/ext/ndb/tasklets.py", line 513, in _on_rpc_completion
    result = rpc.get_result()
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia_tools/google_appengine_1.9.67/google_appengine/google/appengine/api/apiproxy_stub_map.py", line 613, in get_result
    return self.__get_result_hook(self)
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia_tools/google_appengine_1.9.67/google_appengine/google/appengine/api/app_identity/app_identity.py", line 527, in get_access_token_result
    rpc.check_success()
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia_tools/google_appengine_1.9.67/google_appengine/google/appengine/api/apiproxy_stub_map.py", line 579, in check_success
    self.__rpc.CheckSuccess()
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia_tools/google_appengine_1.9.67/google_appengine/google/appengine/api/apiproxy_rpc.py", line 157, in _WaitImpl
    self.request, self.response)
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia_tools/google_appengine_1.9.67/google_appengine/google/appengine/ext/remote_api/remote_api_stub.py", line 222, in MakeSyncCall
    self._MakeRealSyncCall(service, call, request, response)
  File "/Users/hananmufti/Desktop/Projects/opensource/oppia_tools/google_appengine_1.9.67/google_appengine/google/appengine/ext/remote_api/remote_api_stub.py", line 256, in _MakeRealSyncCall
    raise pickle.loads(response_pb.exception())
RuntimeError: AccessTokenRefreshError(u'invalid_grant: Bad Request',)

ERROR    2020-06-11 18:49:00,450 base.py:476] Exception raised: AccessTokenRefreshError(u'invalid_grant: Bad Request',)
INFO     2020-06-11 18:49:00,690 module.py:835] default: "POST /createhandler/imageupload/story/RHhM5bpZumPg HTTP/1.1" 500 93
```

Creating translations works as before!
<img width="1234" alt="Screenshot 2020-06-11 at 20 54 09" src="https://user-images.githubusercontent.com/11369764/84427777-bcd40380-ac25-11ea-90cd-ac7d549b9da5.png">

**PTAL!**